lips/lip-3.md

```diff
@@ -8,52 +8,51 @@ created: 06/09/2020
+In order to release a new version of software, Libra Networks (LN) needs to coordinate the release plan and timeline with all the validator operators. Validator operators are responsible for deploying the software release as per the provided instructions.
 
-Some specification changes may require all validators to transition at the same height, which is accompanied with an on-chain state value update (described in the next section).
+Some changes may require all validators to transition at the same height, which is accompanied with an [on-chain state update](#on-chain-state-upgrade).
 
 ## On-chain state upgrade
-On-chain state can only be updated via transactions. We propose three ways for LN to execute critical updates, ordered by most frequently expected to least frequently expected: normal multi-sig transactions, multi-sig WriteSet transactions, and genesis transactions.
+On-chain state can only be updated via transactions. We propose three ways to execute critical updates, ordered by most frequent to least frequent: normal multi-sig transactions, multi-sig WriteSet transactions, and genesis transactions.
 
 ### Normal multi-sig transactions
-Normal transactions go through VM and smart contracts, both the resulting state change and the authenticating set of signatures are transparent and visible on-chain for auditing.
-
-Different groups would authenticate different types of transactions. For example, there would be a group for compliance transactions and another group for validator set management.
+Normal transactions go through the VM and invoke Libra Framework modules (see [LIP-2](./lip-2.md)) to produce a Writeset that alters ledger state. Both the resulting state change and the authenticating set of signatures are transparent and visible on-chain for auditing.
 
 ### Multi-sig WriteSet transactions
-Writeset transactions are a special type of multi-sig transactions that prescribes an arbitrary state change. Unlike normal transactions, which must follow rules such as “the reserve is the only entity that can create or destroy money”, WriteSet transactions simply describe a state change rather than generate one using the blockchain logic that is encoded in Move. This type of transaction is only needed for extremely rare situations such as updating immutable stdlib libraries due to critical bugs or reversing large scale fraud.
+Writeset transactions are a special type of multi-sig transactions that prescribes an arbitrary state change. Unlike normal transactions, which must follow rules such as “the reserve is the only entity that can create or destroy money”, WriteSet transactions simply describe a state change rather than generate one using the blockchain logic that is encoded in Move. This type of transaction is only needed for situations such as updating immutable stdlib libraries due to critical bugs or reversing large scale fraud.
```

Comment:
"than generate one using the blockchain logic that is encoded in Move" --> than go through the Framework modules/rules.